MonoGame.Framework/Graphics/SpriteFont.cs

```diff
@@ -175,40 +198,88 @@ internal void MeasureString(ref CharacterSource text, out Vector2 size)
                     continue;
                 }
 
-                if (!_glyphs.TryGetValue(c, out currentGlyph))
-                {
-                    if (!defaultGlyph.HasValue)
-                        throw new ArgumentException(Errors.TextContainsUnresolvableCharacters, "text");
-
-                    currentGlyph = defaultGlyph.Value;
-                }
+                var currentGlyphIndex = GetGlyphIndexOrDefault(c, defaultGlyphIndex);
+                var pCurrentGlyph = pGlyphs + currentGlyphIndex;
 
                 // The first character on a line might have a negative left side bearing.
                 // In this scenario, SpriteBatch/SpriteFont normally offset the text to the right,
                 //  so that text does not hang off the left side of its rectangle.
                 if (firstGlyphOfLine) {
-                    offset.X = Math.Max(currentGlyph.LeftSideBearing, 0);
+                    offset.X = Math.Max(pCurrentGlyph->LeftSideBearing, 0);
                     firstGlyphOfLine = false;
                 } else {
-                    offset.X += Spacing + currentGlyph.LeftSideBearing;
+                    offset.X += Spacing + pCurrentGlyph->LeftSideBearing;
                 }
 
-                offset.X += currentGlyph.Width;
+                offset.X += pCurrentGlyph->Width;
 
-                var proposedWidth = offset.X + Math.Max(currentGlyph.RightSideBearing, 0);
+                var proposedWidth = offset.X + Math.Max(pCurrentGlyph->RightSideBearing, 0);
                 if (proposedWidth > width)
                     width = proposedWidth;
 
-                offset.X += currentGlyph.RightSideBearing;
+                offset.X += pCurrentGlyph->RightSideBearing;
 
-                if (currentGlyph.Cropping.Height > finalLineHeight)
-                    finalLineHeight = currentGlyph.Cropping.Height;
+                if (pCurrentGlyph->Cropping.Height > finalLineHeight)
+                    finalLineHeight = pCurrentGlyph->Cropping.Height;
             }
 
             size.X = width;
             size.Y = offset.Y + finalLineHeight;
 		}
 
+        private int GetRegionIndex(char c)
+        {            
+            var l = 0;
+            var r = _regions.Length - 1;
+
+            while (l <= r)
+            {
+                var m = (l + r) >> 1;
+                if (_regions[m].End < c)
+                {
+                    l = m + 1;
+                }
+                else if (_regions[m].Start > c)
+                {
+                    r = m;
+                    if (l == r) return l;
+                }
+                else
+                {
+                    return m;
+                }
+            }
+
+            return -1;
+        }
+
+        internal bool TryGetGlyphIndex(char c, out int index)
+        {   
+            var regionIdx = GetRegionIndex(c);

```

Comment:
Might as well inline this...  no one else calls it.  Plus then you can consider using an unsafe pointer to access `_regions`.